Add package access modifier [SE-0386] (#288)
apple/swift-book: TSPL.docc/ReferenceManual/Declarations.md

  Declarations marked with the `public` access-level modifier can also be accessed (but not subclassed)
  by code in a module that imports the module that contains that declaration.

- term `package`:
  Apply this modifier to a declaration

>   to indicate that the declaration can be accessed
>   only by code in the same package as the declaration.

Reviewer: Change passive voice to active:
```suggestion
  to indicate that only code in the same package 
  as the declaration can access it.
```

Author: Suggest stet.  This wording is parallel to the existing wording for the other access levels.  The focus of this bulleted list is on the declaration, not on what's using it, so these descriptions tend to be easier to read when their wording follows that ordering.

That said, this usage doesn't quite rise to the level of "undefined, unimportant actor" that the TSPL [style guide](https://github.com/apple/swift-book/blob/main/Style.md#passive-voice) notes is a place where the book uses passive voice — there is an actor here, although it's somewhat vaguely defined "other code somewhere else".  So there's room for argument that we should rewrite the entire list.

Reviewer: Oh, thanks for sharing the TSPL style guide! I've bookmarked that for future reference.

The level of effort to update the entire list to be active doesn't seem worth it. I'm ok with leaving as passive to be parallel with the other items.

  A package is a unit of code distribution
  that you define in the build system you're using.
  When the build system compiles code,
  it specifies the package name
  by passing the `-package-name` flag to the Swift compiler.
  Two modules are part of the same package

>   if the build system specifies the same package name when building them both.

Reviewer 2: The package name is set automatically in a typical Swift package but not for an Xcode project. Would it be appropriate to mention the Xcode build setting to set the package name here? For the record, it's `Package Access Identifier`.

Author: I added Xcode as another example next to the SwiftPM example in 8aafa213.  How's that look?

Reviewer 2: That's good to know, I can understand that separation. The change looks fine to me.

>   if the build system specifies the same package name when building them both.

Reviewer: ```suggestion
  if the build system specifies the same package name when building both of them.
```

Author: Changed to omit "both".  While applying the edit, I wasn't quite happy with either the new or old version here, and I think dropping "both" makes it less ambiguous.  The build system doesn't need to build both things at the same time — it just needs to specify the same package name when it builds each of them.
